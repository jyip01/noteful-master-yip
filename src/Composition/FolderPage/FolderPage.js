import React, {Component} from 'react';
import HomeSideBar from '../HomeSideBar/HomeSideBar'
import FolderMain from '../FolderMain/FolderMain'
import ApiContext from '../../ApiContext'
import Error from '../../Error.js'

class FolderPage extends Component {
    static contextType = ApiContext;
        
    //for refreshing
        componentDidMount(){
            if(!this.context.notes.length || !this.context.folders.length || !this.context.currentFolder){
                this.context.fetchData()
            }
        }
    
        componentWillUpdate(nextprops){
           console.log(nextprops)
            if(!this.context.currentFolder){
                       
                const {props: {match}} = nextprops
                const folder = this.context.folders.find(f =>
                f.id === match.params.folderId)
            
            this.context.setFolder(folder)
            }
     
        }
    

    render(){
        if(!this.context.folders.length || !this.context.notes.length || !this.context.currentFolder){
            return null
        }
        
        return(<Error>
            <div className="contentContainer folderPage">
                <HomeSideBar/>
                <FolderMain/>
            </div>
        </Error>
        )
    }
}

export default FolderPage