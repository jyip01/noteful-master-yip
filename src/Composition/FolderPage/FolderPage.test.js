import React from 'react'
import ReactDOM from 'react-dom'
import FolderPage from './FolderPage'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'


it('renders without crashing',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><FolderPage/></BrowserRouter>,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <FolderPage/>
        </BrowserRouter>
      )
      .toJSON();
     expect(tree).toMatchSnapshot();
     });
