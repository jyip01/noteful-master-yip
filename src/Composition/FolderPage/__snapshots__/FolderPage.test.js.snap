// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `null`;
